daemon/image_delete.go

```diff
@@ -132,15 +146,16 @@ func (daemon *Daemon) imgDeleteHelper(name string, list *[]types.ImageDelete, fi
 				if first {
 					return err
 				}
-
 			}
-
 		}
 	}
 	return nil
 }
 
 func (daemon *Daemon) canDeleteImage(imgID string, force bool) error {
+	directlyUsedBy := []string{}
```

Comment:
But, in this case, only the first image is transitive, since we flop that flag on the very first iteration through the loop.